### ISSUE TITLE
Sideloaded app is crashing (even when not injected)

### ISSUE THREAD

**Issue by krjan02** [Rov3r/scmusicplus#9]
I tried to patch (Sideloady)both the decrypted IPA aswell as the IPA from iMazing.

All it does is crash when launching the App.

What am i doing wrong?

Info: iOS 18 (EU Sideloading + Certificate)

**Comment by Rov3r** [comment]
Possibly you tried to use the corrupted rootless release. I fixed it, so try again with the latest release please

**Comment by krjan02** [comment]
I tried patching again with Sideloady and Scarlet (directly on the iPhone). Sadly the app either crashes or - when i open the app - i get a message that the app was not installed correctly and i should try again later.

**Comment by HeyItsMicro** [comment]
I'm also having the same issue. I used a 7.55.0 decrypted SoundCloud ipa, injected it with the latest deb (I tried both), signed it, airdropped it to my phone to install it, and it's crashing on launch. I'm on iOS 18.1 beta 4.

**Comment by Rov3r** [comment]
I believe I have located the issue. For some reason, the app decryption process is not fully decrypting the app. See if you can sideload just the SoundCloud ipa, without SCMusicPlus injected. Please report back if the ipa installs and opens correctly.

**Comment by mkkvcs** [comment]
Joining in on the discussion, no, the decrypted app doesn't open either with or without the injection

**Comment by Rov3r** [comment]
Just as I thought. Seems to be an issue with the SoundCloud ipa itself. Something along the line is not being decrypted. I got this error when trying to install on TrollStore
![trollstore_error](https://github.com/user-attachments/assets/4e5c358e-86f7-4263-9dc1-0feac5186ee7)

I will contact some people and try to get this sorted. In the meantime it will just not work on sideloaded installations. It seems to still work on TrollStore installs though (possibly with the caveat that the decrypted ipa is from the same device it was installed on)

**Comment by Rov3r** [comment]
I'll need someone who has a rootful jailbroken device on iOS 15 to decrypt and email it to me or test if it sideloads without error. I don't have a device capable of decrypting via iridium on the latest version.

**Comment by krjan02** [comment]
I have tried to decrypt the app with Irdium on a paler1n rootful iPhone 6s Plus iOS 15.6.
[Iridium.txt](https://github.com/user-attachments/files/17114041/Iridium.txt)

Also same message (... some parts encrypted... ) when installing with TrollStore (on the rootful Device) .

**Comment by Rov3r** [comment]
Thanks for testing it out! I’ll continue trying to figure out why this is.

**Comment by c12n** [comment]
There's an already an injected ipa decrypted with the tweak in it. 

t.me/SultanModsIPAs

**Comment by KaaaooS** [comment]
> There's an already an injected ipa decrypted with the tweak in it.
> 
> t.me/SultanModsIPAs

Thats right,and say "ads not removed". Maybe 24.9.1-1 version not working fine at all with the ad block??

**Comment by Rov3r** [comment]
> Thats right,and say "ads not removed". Maybe 24.9.1-1 version not working fine at all with the ad block??

Incorrect. If you read this issue, sideloading even the unmodified ipa will result in a crash upon launch. And you must use DNS filter to block remaining ads.

**Comment by KaaaooS** [comment]
> > Thats right,and say "ads not removed". Maybe 24.9.1-1 version not working fine at all with the ad block??
> 
> Incorrect. If you read this issue, sideloading even the unmodified ipa will result in a crash upon launch. And you must use DNS filter to block remaining ads.

And that problem with ads its for the unmodified ipa?? I remember with last version is working so fine with no dns

**Comment by Rov3r** [comment]
> And that problem with ads its for the unmodified ipa?? I remember with last version is working so fine with no dns

This issue is not pertaining to ad blocking functionality. Comment on the issue you already made about it if you want to continue this discussion.

**Comment by DirkBraakman** [comment]
24.9.1-3, sideloaded with Sideloadly, iPhone 15 HK (EU OS), iOS 18, SoundCloud Decrypted 7.56.0, arm64 Tweak:
App still crashes immediately after launching.

Also, if people have problems with a mismatched bundle ID with your decrypted IPA:
1. Note down expected ID from Sideloadly error
2. Download Xplist: https://github.com/ic005k/Xplist
3. Download and install 7-Zip: https://www.7-zip.org/
4. Extract downloaded decrypted IPA
5. With Xplist, edit "com.soundcloud.TouchApp-X.XX.X-Decrypted\Payload\SoundCloud.app\Watch\SCWatch.app\Info.plist"
6. Edit "WKCompanionAppBundleIdentifier" to: com.soundcloud.TouchApp.(Expected ID From Sideloadly)
7. Edit "CFBundleIdentifier" to: com.soundcloud.TouchApp.(Expected ID From Sideloadly).watchkitapp
8. Save with Xplist
9. Archive "com.soundcloud.TouchApp-X.XX.X-Decrypted\Payload" and "\iTunesArtwork" together as .ZIP file
10. Rename .ZIP file to .IPA file (com.soundcloud.TouchApp-X.XX.X-Decrypted.ipa)
11. Sideload using Sideloadly and make sure to inject SCMusicPlus .DEB tweak

**Comment by Rov3r** [comment]
Something must have been updated with Sideloadly and/or the decryption tools because I'm now able to use TrollDecrypt to grab the IPA and then Sideloadly's Sideload Spoofer and have no issues signing in or launching the app.
